src/main/scala/firrtl/passes/AnnotateMemMacros.scala

```diff
@@ -44,35 +44,45 @@ object AnalysisUtils {
   // use case: compare if two nodes have the same origin
   // limitation: only works in a module (stops @ module inputs)
   // TODO: more thorough (i.e. a + 0 = a)
-  def getConnectOrigin(connects: Connects, node: String): Expression = {
-    if (connects contains node) getOrigin(connects, connects(node))
-    else EmptyExpression
-  }
+  def getConnectOrigin(connects: Connects)(node: String): Expression =
+    connects get node match {
+      case None => EmptyExpression
+      case Some(e) => getOrigin(connects, e)
+    }
+  def getConnectOrigin(connects: Connects, e: Expression): Expression =
+    getConnectOrigin(connects)(e.serialize)
 
   private def getOrigin(connects: Connects, e: Expression): Expression = e match { 
     case Mux(cond, tv, fv, _) =>
       val fvOrigin = getOrigin(connects, fv)
       val tvOrigin = getOrigin(connects, tv)
       val condOrigin = getOrigin(connects, cond)
-      if (we(tvOrigin) == we(one) && we(fvOrigin) == we(zero)) condOrigin
-      else if (we(condOrigin) == we(one)) tvOrigin
-      else if (we(condOrigin) == we(zero)) fvOrigin
-      else if (we(tvOrigin) == we(fvOrigin)) tvOrigin
-      else if (we(fvOrigin) == we(zero) && we(condOrigin) == we(tvOrigin)) condOrigin
+      if (weq(tvOrigin, one) && weq(fvOrigin, zero)) condOrigin
+      else if (weq(condOrigin, one)) tvOrigin
+      else if (weq(condOrigin, zero)) fvOrigin
+      else if (weq(tvOrigin, fvOrigin)) tvOrigin
+      else if (weq(fvOrigin, zero) && weq(condOrigin, tvOrigin)) condOrigin
       else e
-    case DoPrim(PrimOps.Or, args, consts, tpe) if args.contains(one) => one
-    case DoPrim(PrimOps.And, args, consts, tpe) if args.contains(zero) => zero
+    case DoPrim(PrimOps.Or, args, consts, tpe) if args exists (weq(_, one)) => one
+    case DoPrim(PrimOps.And, args, consts, tpe) if args exists (weq(_, zero)) => zero
     case DoPrim(PrimOps.Bits, args, Seq(msb, lsb), tpe) =>  
-      val extractionWidth = (msb-lsb)+1
+      val extractionWidth = (msb - lsb) + 1
       val nodeWidth = bitWidth(args.head.tpe)
       // if you're extracting the full bitwidth, then keep searching for origin
-      if (nodeWidth == extractionWidth) getOrigin(connects, args.head)
-      else e
+      nodeWidth == extractionWidth match {
```

Comment:
Especially for something like this when there are only 2 conditions, doing a match for true/false seems overkill o_o